Apps/SurfaceStatus/SurfaceStatusApp/AdapterPayloads/codex-presence-launcher.py

```
@@ -0,0 +1,156 @@
+    env.pop("CMUX_CODEX_HOOKS_DISABLED", None)
+    env["CMUX_CODEX_PID"] = str(os.getpid())
+    env["CMUX_AGENT_LAUNCH_KIND"] = "codex"
+    env["CMUX_AGENT_LAUNCH_EXECUTABLE"] = str(codex)
+    env["CMUX_AGENT_LAUNCH_CWD"] = os.getcwd()
+
+    surface_id = valid_uuid(env.get("CMUX_SURFACE_ID", ""))
+    workspace_id = valid_uuid(env.get("CMUX_WORKSPACE_ID", ""))
+    if surface_id is not None:
+        publish_marker(pathlib.Path.home().resolve(), surface_id, workspace_id)
+
+    os.execve(codex, args, env)
```

Ruff (0.16.0)

[error] Starting a process without a shell

(S606)